VACMS-21548: AI UI changes

omahane opened the pull request:
## Description
Updates the UI/UX of adding and regenerating alt text for images.

Relates to #21548 

### Generated description
This pull request introduces changes to enhance the user experience and accessibility of media-related features, as well as minor adjustments to styling and test behavior. The key updates include adding functionality for AI-generated alt text regeneration, improving accessibility descriptions, updating styles for media widgets, and modifying test wait times.

### Media functionality and accessibility improvements:
* [`docroot/modules/custom/va_gov_media/va_gov_media.module`](diffhunk://#diff-e9b22afa9b59d7298883dff52dcf067433b367cd296fb35bccda83827f508aa7R85-R100): Added a new implementation of `hook_field_widget_single_element_form_alter()` to update the button text for regenerating alt text and enhance the alt field description for better accessibility.

### Styling updates:
* [`docroot/themes/custom/vagovclaro/assets/scss/components/_media.scss`](diffhunk://#diff-709ea810f7790327d61227b2d61efddc4ee522ac9520716f9effd2bc13650094R89-R94): Added a CSS rule to unset text alignment for the `.ai-alt-text-generation-wrapper` inside the `.media-library-widget-modal`.

### Template adjustments:
* [`docroot/themes/custom/vagovclaro/templates/content-edit/file-managed-file.html.twig`](diffhunk://#diff-4fc61ff34f72a1f19cc2df5c3d3f747c102e8e8e3f82e3ae3fa4b79a3cbc4305L67-R67): Removed the `hide` class from the inline guidance text box, making it visible.

### Test behavior updates:
* [`tests/cypress/integration/step_definitions/common/i_create_a_media.js`](diffhunk://#diff-03720ecb1418bd1363649ac0dacd405cb6942f217a908b70b5b50ee5d1967e66L57-R57): Increased the wait time from 1000ms to 3000ms to ensure stability during media-related test execution.

## Testing done
Manually

## Screenshots
<img width="1680" height="1050" alt="Screenshot 2025-08-05 at 2 05 08 PM" src="https://github.com/user-attachments/assets/9600e0e9-e70b-423c-91da-0987d19a888a" />

### Screenshot with large portrait image
_Note: The height of the images in the preview are now capped at 300 px to make it easier to preview them and see regenerate button._
<img width="1681" height="1137" alt="Screenshot 2025-08-18 at 2 53 46 PM" src="https://github.com/user-attachments/assets/56e3664f-7362-420c-b3b7-306b87487def" />

### Screenshot with landscape image
<img width="1681" height="1137" alt="Screenshot 2025-08-18 at 2 56 46 PM" src="https://github.com/user-attachments/assets/d7a7d989-4942-4571-8553-5d955d4b347d" />




## QA steps
### Update the QA Content Publisher user
- [x] Assign QA Content Publisher to the following section:
  - [x] VA Boston health care
- [x] Assign QA Content Publisher to the following roles:
  - [x] Content creator - VAMC
  - [x] Content publisher

### Create a story
- [x] Log in as [QA Content Publisher](https://pr21874-8pj7uaejilixh8qgnamdkzbhbm81wwzf.ci.cms.va.gov/)
- [x] [Add a story](https://pr21874-8pj7uaejilixh8qgnamdkzbhbm81wwzf.ci.cms.va.gov/node/add/news_story)
- [x] Add an image
- [x] After the image uploads
  - [x] Confirm that the alt text is automatically generated
  - [x] Confirm that the alt text tips are open
  - [x] Confirm that the alt text help text is as follows: Short description of the image used by screen readers and displayed when the image is not loaded. This text is important for accessibility. Verify that the autogenerated text meaningfully represents the content of the image. If necessary, edit the text directly within the field or regenerate new text by selecting the "Regenerate alt text" button.'
  - [x] Confirm that the button is labeled "Regenerate alt text"
  - [x] Use the button to regenerate alt text
  - [x] Confirm that the alt text changes
  - [x] Edit the alt text manually
  - [x] Save and insert the image
- [x] Finish adding the rest of the story fields
- [x] Save the story
- [x] In node:view, open the developer tools
- [x] Confirm that the alt text matches the text you edited

### Change a facility image
- [x] Edit the Jamaica Plains Medical Center
- [x] Remove the facility photo
- [x] Replace it with another photo
- [x] After the image uploads
  - [x] Confirm that the alt text is automatically generated
  - [x] Confirm that the alt text tips are open
  - [x] Confirm that the alt text help text is as follows: Short description of the image used by screen readers and displayed when the image is not loaded. This text is important for accessibility. Verify that the autogenerated text meaningfully represents the content of the image. If necessary, edit the text directly within the field or regenerate new text by selecting the "Regenerate alt text" button.'
  - [x] Confirm that the button is labeled "Regenerate alt text"
  - [x] Use the button to regenerate alt text
  - [x] Confirm that the alt text changes
  - [x] Edit the alt text manually
  - [x] Save and insert the image

### Select Team for PR review

- [x] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

laflannery commented:
@omahane  My first question, I don't see the AI button, so that's curious. I'm here https://pr21874-8pj7uaejilixh8qgnamdkzbhbm81wwzf.ci.cms.va.gov/media/add/image and I see this:
<img width="1818" height="461" alt="Screenshot 2025-07-31 at 9 23 05 AM" src="https://github.com/user-attachments/assets/df1bc54a-ba95-4af6-92df-efa5efdfa3e4" />

However, the one comment I do have with what I am seeing is that the "Tips for Alternative text" trigger, while visually it is default to open, programmatically it is not which is going to confuse things. By that I mean, that when it first loads the `aria-expanded` attribute is still set to false. You can tell it is by looking at the code obviously but also you can tell be cause the caret is pointing the wrong way, when expanded the caret points up, when collapsed the caret points down:

<img width="1193" height="406" alt="Screenshot 2025-07-31 at 9 25 49 AM" src="https://github.com/user-attachments/assets/b29c3023-d352-4965-b067-08269e1d8a51" />

omahane commented:
> @omahane My first question, I don't see the AI button, so that's curious. 

This was a permissions issue. I have added permission to use this button to a variety of roles.

> However, the one comment I do have with what I am seeing is that the "Tips for Alternative text" trigger, while visually it is default to open, programmatically it is not which is going to confuse things. 

I have updated the template to make `aria-expanded` default to true.

laflannery reviewed (approved):
Permissions look good now and the aria attribute is working correctly!
